### PR TITLE
Remove libgit2 and git2go from codebase

### PR DESCRIPTION
**Pull request by pjbgf**
This PR removes the `libgit2` implementation which was deprecated in [v0.27.0](https://github.com/fluxcd/image-automation-controller/blob/main/CHANGELOG.md#0270),
resulting in a smaller code base and more streamlined build process.

Test image: `ghcr.io/fluxcd/image-automation-controller:rc-95ace68b`.

Things of note:
- Some of the fuzz changes were orthogonal and related to https://github.com/fluxcd/flux2/issues/3346.
- The generated container images still contain a statically built controller binary.

Breaking changes:
- Feature gate `ForceGoGitImplementation` has been deprecated and removed.

### CODE REVIEW

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @pjbgf 🏅

File: docs/spec/v1beta1/imageupdateautomations.md [hunk 1]
```diff
@@ -111,9 +111,8 @@ To be able to commit changes back, the referenced `GitRepository` object must re
 with write access; e.g., if using a GitHub deploy key, "Allow write access" should be checked when
 creating it. Only the `url`, `ref`, and `secretRef` fields of the `GitRepository` are used.
 
-The [`gitImplementation` field][source-docs] in the referenced `GitRepository` is ignored. The
-automation controller automatically uses the go-git implementation rather than libgit2.
-To change this behavor, start the controller with `--feature-gates=ForceGoGitImplementation=false`.
+The [`gitImplementation` field][source-docs] in the referenced `GitRepository` is now ignored. All
+reconciliations are now executed using the `go-git` implementation.
```
Comment (hiddeco) [inline docs/spec/v1beta1/imageupdateautomations.md:115]:
```suggestion
The [`gitImplementation` field][source-docs] in the referenced `GitRepository` is ignored. All
reconciliations are executed using the `go-git` implementation.
```

File: Dockerfile [hunk 2]
```diff
@@ -17,12 +12,18 @@ FROM gostable AS go-linux
...
-# build-go-mod can still be cached at build platform architecture.
-FROM build-base as build-go-mod
+# build can still be cached at build platform architecture.
+FROM build-base as build
+
+ARG TARGETPLATFORM
+
+# Some dependencies have to installed
+# for the target platform: https://github.com/tonistiigi/xx#go--cgo
+RUN xx-apk add musl-dev gcc
```
Comment (hiddeco) [inline Dockerfile:26]: Ref recommends explicit declaration of `CGO_ENABLED`.

In addition: might be worth taking note of a brief "why" for the CGO dependency.

**Review by hiddeco — COMMENTED** [review]
🙌 awesome work coming together @pjbgf (& @aryan9600)

Two tiny nits.

**Comment by pjbgf** [comment]
Successfully tested the rc image against:
- Azure DevOps
- GitLab push branch
- BitBucket
- GitHub
- GitHub push branch

**Pull request merged**